File: ruby/lib/medusa.rb
class Medusa
  attr_reader :name, :statues

  def initialize(name)
    @name = name
    @statues = []
  end

  def stare(victim)
    @statues << victim
    victim.becomes_stone
    @statues.shift.become_human if @statues.count > 3
  end
end


class Person
  attr_reader :name

  def initialize(name)
    @name = name
    @stoned = false
  end

  def stoned?
    @stoned
  end

  def becomes_stone
    @stoned = true
  end

  def become_human
    @stoned = false
  end
end

File: javascript/exercises/medusa.js
class Medusa {
  constructor(name){
    this.name = name
    this.statues = []
  }

  stare(victim){
    if (this.statues.length < 3) {
      this.statues.push(victim)
      victim.becomes_stone();
    }
    else if (this.statues.length === 3){
      this.statues[0].becomes_human()
      this.statues.shift();
      victim.becomes_stone();
      this.statues.push(victim);
    }
  };
};

module.exports = Medusa;